src/util.ts

```diff
@@ -0,0 +1,153 @@
+export type Metadata = {
+	size: number,
+	keyHash: Buffer
+}
+export type FileOptions = { 
+	encoding?: BufferEncoding | undefined, 
+	mode?: number | undefined, 
+	flag?: string | undefined
+}
+export type WriteFileOptions = { 
+	encoding?: BufferEncoding | undefined,
+	mode?: number | undefined,
+	flag?: string | undefined
+}
+export type PathLike = string | Buffer | URL
+export type FileLike = PathLike | number
+export type NoParamCallback = (err: NodeJS.ErrnoException | null) => void
+export interface RmDirOptions {
+	recursive?: boolean
+}
+export interface MakeDirectoryOptions {
+	recursive?: boolean,
+	mode?: number
+}
+export interface Stat {
```

Comment:
We should get this from the `"@node/types"`. So this shouldn't exist just like what we talked about with respect to the `GeneralFilesystem`.

Reply:
These redundant interfaces have been removed and replaced with types from nodes.